src/linux/system.rs

```diff
@@ -487,6 +487,15 @@ impl SystemExt for System {
         &self.users
     }
 
+    #[cfg(not(target_os = "android"))]
+    fn get_name(&self) -> Option<String> {
+        get_system_info(
+            InfoType::Name,
+            Some(BufReader::new(File::open("/etc/os-release").ok()?)),
```

Comment:
That's still not what I asked haha. Why do we want so much to give a `File` or a `Buffer` directly to the `get_system_info` function? Do you have a reason for it?

Reply:
Yep I'm still working on this, haven't updated it here yet.   Will do shortly.  The main reason I gave a buffer was to make testing easier.  I can also give a string directly as suggested.  If I left the signature as before the new functionality can't be tested directly unless the test is run on a machine that is missing that field in `os-release`.

Reply:
Oh I see. Just create a temp file then instead. ;)